### PR TITLE
#22365  initial commit

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
Adding CT Push Pull functionality to the CLI

### CODE REVIEW

**Review by wezell — COMMENTED** [review]
Please add a README.md on how to build the cli

File: tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/contenttype/model/type/ContentType.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    public abstract Date iDate();
+
+    public abstract String host();
+
+    public abstract String folder();
+
+    @Nullable
+    public abstract String icon();
+
+    public abstract int sortOrder();
```
Comment (wezell) [inline tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/contenttype/model/type/ContentType.java:62]: not needed

File: tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/contenttype/model/type/ContentType.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    @Nullable
+    public abstract String description();
+
+    @Nullable
+    public abstract Boolean defaultType();
+
+    public abstract BaseContentType baseType();
+
+    @Nullable
+    public abstract Boolean versionable();
```
Comment (wezell) [inline tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/contenttype/model/type/ContentType.java:73]: not needed

File: tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/contenttype/model/type/ContentType.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    @Nullable
+    public abstract Boolean system();
+
+    @Nullable
+    public abstract String owner();
+
+    @Nullable
+    public abstract Boolean multilingualable();
```
Comment (wezell) [inline tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/contenttype/model/type/ContentType.java:82]: not needed

**Review by wezell — CHANGES_REQUESTED** [review]
Please add a README.md on how to build the cli

**Comment by fabrizzio-dotCMS** [comment]
> Please add a README.md on how to build the cli

done. Readme's in

**Review by jdotcms — APPROVED** [review]

**Review by spbolton — APPROVED** [review]

**Pull request merged**